protocol/communities/manager.go

```diff
@@ -167,6 +158,47 @@ type HistoryArchiveDownloadTask struct {
 	Cancelled  bool
 }
 
+type HistoryArchiveDownloadTaskInfo struct {
+	TotalDownloadedArchivesCount int
+	TotalArchivesCount           int
+	Cancelled                    bool
+}
+
+type ArchiveContract interface {
+	CreateHistoryArchiveTorrentFromMessages(communityID types.HexBytes, messages []*types.Message, topics []types.TopicType, startDate time.Time, endDate time.Time, partition time.Duration, encrypt bool) ([]string, error)
+	CreateHistoryArchiveTorrentFromDB(communityID types.HexBytes, topics []types.TopicType, startDate time.Time, endDate time.Time, partition time.Duration, encrypt bool) ([]string, error)
+	SaveMessageArchiveID(communityID types.HexBytes, hash string) error
+	GetMessageArchiveIDsToImport(communityID types.HexBytes) ([]string, error)
+	SetMessageArchiveIDImported(communityID types.HexBytes, hash string, imported bool) error
+	ExtractMessagesFromHistoryArchive(communityID types.HexBytes, archiveID string) ([]*protobuf.WakuMessage, error)
+	GetHistoryArchiveMagnetlink(communityID types.HexBytes) (string, error)
+	LoadHistoryArchiveIndexFromFile(myKey *ecdsa.PrivateKey, communityID types.HexBytes) (*protobuf.WakuMessageArchiveIndex, error)
+}
+
+type TorrentContract interface {
```

Comment:
This is reasonable. And on the subject of why `Archive` doesn't contain `Torrent` it is basically because of the order in which I did the code refactor. See https://github.com/status-im/status-go/pull/5257#issuecomment-2152326864

Reply:
https://github.com/status-im/status-go/pull/5257/commits/6cffdf324a6bf2c5ac289cef6b37de5518becb99